modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/artifact/ArtifactContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal.artifact;
+
+import org.mule.runtime.config.spring.MuleArtifactContext;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.connectivity.ConnectivityTestingService;
+
+/**
+ * An artifact context contains all the information related to an {@link org.mule.runtime.module.artifact.Artifact}
+ * that contains configuration.
+ *
+ * This object holds the created {@link MuleContext} and the {@link MuleArtifactContext} which holds the information
+ * related to the application configuration and resources.
+ *
+ * @since 4.0
+ */
+public class ArtifactContext {
+
+  private MuleContext muleContext;
+  private MuleArtifactContext muleArtifactContext;
+
+  /**
+   * Creates an {@link ArtifactContext}.
+   *
+   * @param muleContext the mule context
+   * @param muleArtifactContext the artifact context.
+     */
+  public ArtifactContext(MuleContext muleContext, MuleArtifactContext muleArtifactContext) {
```

Comment:
I may be able to remove the MuleContext parameter. I don't want to expose directly the MuleArtifactContext because is a ApplicationContext from spring. I don't even want to expose the MuleContext but I can't change that for now.